new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,73 @@
+import {downloadBlob} from "../utils/file";
+
+
+/**
+ * Manager for exporting logs to a file.
+ */
+class LogExportManager {
+    /**
+     * Array to store chunks of log data.
+     *
+     * @type {string[]}
+     * @private
+     */
+    readonly #chunks: string[] = [];
+
+    /**
+     * Total number of chunks to export.
+     *
+     * @type {number}
+     * @private
```

Comment:
1. The type in JSDoc can be omitted since the types are defined with TypeScript.
2. No need to mark `#` attributes as `@private` as a convention.
```suggestion
```